src/vocabulary.js

```diff
@@ -142,4 +121,48 @@ function getVocabulary() {
   return _vocabulary;
 }
 
+function generateDynamicCaseAttrs(numOfAttrs, numOfUniqueValues){
+  const dynamicDict = {};
+  const str = "Dynamic_";
+  for (let i = 1; i <= numOfAttrs; i++) {
+    const uniqueWords = new Set();
+    while (uniqueWords.size < numOfUniqueValues) {
+      uniqueWords.add(faker.word.sample({ length: { min: 8, max: 15 }, strategy: "closest" }));
+    }
+    dynamicDict[`${str}${i}`] = Array.from(uniqueWords);
+  }
+  return dynamicDict;
+}
+
+function insertDynamicAttributes(items) {
+  const dynamicCaseAttrs = generateDynamicCaseAttrs(config.DYNAMIC_ATTRS, config.UNIQUE_VALUES_FOR_DYNAMIC_ATTRS);
+
+  Object.entries(dynamicCaseAttrs).forEach(([key, value]) => {
+    // Insert new columns into the schema
+    const newColumn = {
+      name: key.toLowerCase(),
+      display_name: key.replace(/_/g,""),
+      type: "varchar(50)",
+      nullable: true
+    };
+    items["__schema__"]["cases"]["columns"].push(newColumn);
+
+    // Calculate weight for each value
+    let weight = 1;
+    if (value.length > 0) {
+      weight = 1 / value.length;
+    }
+
+    // Create a dict with attr values as keys and weights as values
+    const valuesWithWeight = {}
+    for (let i = 0; i < value.length; i++) {
+      valuesWithWeight[value[i]] = weight;
+    }
+    console.log(valuesWithWeight);
```

Comment:
Apologies. I should've been more specific. I think that the dynamic data generation is most suited to be placed in the data.js file since that's the place where the other data is generated. This file is primarily used to parse the vocabulary. A better place to generate the dynamic data is within the `generateCases()` function in data.js.

Reply:
As discussed, renamed the function to `includeDynamicAttributes()`. Function definition lives in vocabulary.json since this function is including dynamic attributes to the vocabulary json and not actually generating any data.